### PR TITLE
Glossary updates

### PR DESCRIPTION
**Pull request by Bosch-0**
Did some formatting on the glossary page as well as edited many entries and added a few new ones. Still lots to be done but a WIP nonetheless! 

Changed the heading size to fit the other pages (###). Some of the longer headings were pretty bulky and looked messy using the older size (##). This page is going to be quite big eventually so saving some space like this makes things a lot cleaner. See difference below old / new: 

Frame 34

### CODE REVIEW

File: guide/glossary.md [hunk 1]
```diff
@@ -160,36 +183,70 @@ The path to the first address in a bitcoin-wallet using BIP84 will look like thi
...
+
+`02f062b2649e0665f2e8e5573172ef85921fc4b307e9795113930ed2229d271f4e`
+
+**Extended public key (xpub)**<br/>
+
+In hierarchical deterministic (HD) wallets, all addresses and their matching public keys are derived from this XPRIV. All public keys and subsequent addresses (by hashing these public keys) are generated from it.
+
+`xpub6DE1bvvTaMkDaAghedChtufg3rDPeYdWt9sM5iTwBVYe9X6bmLenQrSexSa1qDscYtidSMUEo9u7TuXg48Y3hBXQr7Yw8zUaEToH1rVgvif`
+
+**ypub** - A xpub belonging to a wallet deriving P2WPKH in P2SH addresses.
```
Comment (GBKS) [inline guide/glossary.md:224]: I find this super hard to parse (for example, `deriving P2WSH in P2SH addresses`). And not sure what I would do with this as a designer. Is this the point where we keep it simple and refer to external, more technical documentation? Could we just say that there are variations of xpubs created by different projects and it's important to notify users about any potential issues they need to be aware of... or something like that?

File: guide/glossary.md [hunk 2]
```diff
@@ -241,46 +294,39 @@ A small piece of data that has all the information needed to generate a specific
...
-## Public key
-A bitcoin address' public key can be derived from the private key. The address itself is a hash of the public key.
+Many wallet-applications work with HD Wallets and recovery phrases, and are interoperable, meaning you can change the application that can control your wallet should you wish (although there are some caveats depending on if they support just BIP32 or also BIP44).
 
-## Replace-by-fee (RBF)
-A node policy that allows an unconfirmed transaction to be replaced with a different transaction that spends at least one of the same inputs and which pays a higher transaction fee. This can sometimes be useful if a transaction has got *stuck* during times of higher-than-normal network-fees.
+### Remote procedure calls (RPC)
 
-## Recovery phrase
+A remote procedure calls (RPC) is when a command is sent to a remote system where it's processed and results are returned. Bitcoin uses RPC to connect to running instances of bitcoind and issue commands such as sending transactions or querying the Bitcoin block chain.
```
Comment (GBKS) [inline guide/glossary.md:307]:
```suggestion
A remote procedure calls (RPC) is when a command is sent to a remote system where it's processed and results are returned. The Bitcoin Core application uses RPC to connect to running instances of bitcoind and issue commands such as sending transactions or querying the Bitcoin block chain.
```
Good to clarify that the Bitcoin Core implementation of Bitcoin uses this (and not somehow Bitcoin in general).

File: guide/glossary.md [hunk 3]
```diff
@@ -241,46 +294,39 @@ A small piece of data that has all the information needed to generate a specific
...
+### Replace-by-fee (RBF)
+A node policy that allows an unconfirmed transaction to be replaced with a different transaction that spends at least one of the same inputs and which pays a higher transaction fee. This can sometimes be useful if a transaction has got *stuck* during times of higher-than-normal network-fees.
 
-Many wallet-applications work with HD Wallets and recovery phrases, and are interoperable, meaning you can change the application that can control your wallet should you wish (although there are some caveats depending on if they support just BIP32 or also BIP44).
 
 **Technicalities** - Recovery of multisig-wallets needs both the extended public key and the recovery phrase of all paticipating keys as well as the master key fingerprint as defined by BIP32 concatenated with the derivation path of the public key. The derivation path is represented as 32-bit little endian unsigned integer indexes concatenated with each other. The number of 32 bit unsigned integer indexes must match the depth provided in the extended public key.
```
Comment (GBKS) [inline guide/glossary.md:315]: Is this technicalities copy in the right place now?

**Comment by GBKS** [comment]
Thanks for taking this on. If you're still working on it, maybe change it to a draft PR?

**Comment by Bosch-0** [comment]
> Thanks for taking this on. If you're still working on it, maybe change it to a draft PR?

For now I am not, this was just some additions I was making when editing the onboarding section for Conor and may have got a little carried away ...

File: guide/glossary.md (hunk 1 above)
Comment (Bosch-0) [reply guide/glossary.md:224]: I was (or maybe someone else can) elaborate on these further. For now I just put that in as a placeholder. I guess including it was merely to make designers aware that these xpub derivatives exist and are case sensitive.

File: guide/glossary.md (hunk 2 above)
Comment (Bosch-0) [reply guide/glossary.md:307]: Good call, added

File: guide/glossary.md (hunk 3 above)
Comment (Bosch-0) [reply guide/glossary.md:315]: No it is not, will rebase - personally I'd just remove this. I'm more for simple details in the glossary and link to more technical explanations externally or link to sections in the Guide that put the term in context like this.

**Comment by GBKS** [comment]
@Bosch-0 are you still working on this? Johns is also planning glossary changes and I want to avoid messy merge conflicts.

**Comment by Bosch-0** [comment]
I will close this one and fix https://github.com/BitcoinDesign/Guide/pull/286
